js/src/components/paid-ads/asset-group/asset-group.js

```diff
@@ -34,9 +34,17 @@ export default function AssetGroup( { campaign } ) {
 	// TODO: When editing, it needs to distinguish whether the given asset group is empty. Will be implemented later.
 	const isEmptyAssetGroup = true;
 	const { isValidForm, handleSubmit, adapter } = useAdaptiveFormContext();
-	const { isSubmitting, isSubmitted, submitter } = adapter;
+	const { isValidCampaign, isSubmitting, isSubmitted, submitter } = adapter;
 	const currentAction = submitter?.dataset.action;
 
+	const handleLaunchClick = ( ...args ) => {
+		if ( isValidForm ) {
+			handleSubmit( ...args );
+		} else {
+			adapter.showValidation();
+		}
+	};
```

Comment:
❓  Maybe to discuss with UX if it would be convenient to show the validation after selecting URL and populate assets the first time?

Reply:
@puntope, thanks for the suggestion.

> Maybe to discuss with UX if it would be convenient to show the validation after selecting URL and populate assets the first time?

> However, I'm thinking if it would be better to show the issues in the very beginning instead of when the user performs a change or clicks on Launch Campaign. I guess this is more a UX question (I didn't check if there is any discussion about in P2) - Quoted from https://github.com/woocommerce/google-listings-and-ads/pull/1863#issuecomment-1403491807

The flow of showing error messages can be found in this design walkthrough video: pcTzPl-1eS-p2#design-walkthrough

https://user-images.githubusercontent.com/17420811/215654469-4357d7bc-b6ef-4f2a-ae63-3b87773cd28e.mp4

The common ways of handling form validation errors are:

1. Errors are displayed after the user leaves an input field.
2. Errors are displayed after the user submits the form.
3. Start displaying errors while the user is entering.

While it's not best practice to display errors after clicking the submit button, displaying errors before the user has entered data or has touched any input fields is considered a bad design (aka premature validation) in many UX recommendations. The reasons for avoiding overly aggressive premature validation are:

- Users may feel frustrated and unfairly reprimanded when a site claims they’ve made mistakes before they’ve had a chance to enter a valid input.
- Users may be distracted from their typing.
- The first glance of the form for users is a bunch of errors that may be very scary.

Therefore, I think it's probably not necessary to discuss this approach with the designer. But if you think this is still the better way to go, I will still discuss it with the designer.